### ISSUE TITLE
Rearranging tiled windows with mouse is awkward due to timing dependence

### ISSUE THREAD

**Issue by peterkrull** [pop-os/cosmic-comp#711]
When initiating a tile-rearrange with the mouse using the Super + Left-click drag, there seems to be a time-delay before windows start to rearrange. This means it is impossible to quickly rearrange windows just by a quick flick of the mouse. Another issue is, even after the timeout, the mouse must move a tiny amount, before the rearrange triggers. If the mouse is moved and held still, it will just return to its original position.

https://github.com/user-attachments/assets/e54d1f05-ac75-465f-93fc-c4f0c8ae0b5d

Is this delay necessary? If it is, could it be made configurable/disable-able?

**Comment by peterkrull** [comment]
I have found that removing the overdue check makes this problem significantly better:
https://github.com/pop-os/cosmic-comp/blob/master/src/shell/layout/tiling/mod.rs#L3548-L3564

And it makes the animations more responsive. Perhaps the animations can be visually a bit more jarring, but I think it is a good tradeoff for the sake of responsiveness and usability.

https://github.com/user-attachments/assets/db4cfb72-7d67-4a47-8a02-cfaeb19bb572

**Comment by romanstingler** [comment]
yes moving / rearranging windows via the mouse should be significantly faster.
Any tiling manager like hyprland,sway it is butter smooth.

**Comment by peterkrull** [comment]
To add some more context. Not only is the long amount of time an issue to me, but the fact that I have to move the cursor _after_ the delay really just feels broken. Perhaps at the end of the relevant delay, an event is emitted to check the cursor position, even if it has not been updated after the ended delay.

**Comment by l-const** [comment]
Could this be addressed somehow?
